vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -567,6 +586,9 @@ void AdsImpl::set(privacy::TokenGeneratorInterface* token_generator) {
   promoted_content_ad_ = std::make_unique<PromotedContentAd>();
   promoted_content_ad_->AddObserver(this);
 
+  search_result_ad_ = std::make_unique<SearchResultAd>();
+  search_result_ad_->AddObserver(this);
```

Comment:
I didn't find where RemoveObserver is called

Reply:
Fixed